### ISSUE TITLE
Homebrew version string

### ISSUE THREAD

**Issue by wyardley** [hashicorp/terraform#32963]
### Terraform Version

```shell
Terraform v1.4.3-dev
on darwin_arm64

Your version of Terraform is out of date! The latest version
is 1.4.4. You can update by downloading from https://www.terraform.io/downloads.html
```

### Terraform Configuration Files

n/a

### Debug Output

n/a

### Expected Behavior

Terraform 1.4.4 should report its version as 1.4.4

### Actual Behavior

Terraform 1.4.4 (when installed via homebrew) reports its version as 1.4.3

### Steps to Reproduce

1. `brew install terraform`
1. `terraform -version`

### Additional Context
https://discuss.hashicorp.com/t/homebrew-version-string/52172
I verified that binary download seems to have the correct version string.

Another issue (can file a bug here as well if needed) is that Ubuntu packages seem to be missing
https://discuss.hashicorp.com/t/ubuntu-download-not-working-for-1-4-3-1-4-4/52133

**Comment by maxb** [comment]
This seems to be because the version.go file in the `v1.4.4` tagged commit, has not been updated: https://github.com/hashicorp/terraform/blob/v1.4.4/version/version.go

**Comment by crw** [comment]
Thanks for this report! Just to set expectations, due to the nature of this problem it may have to wait for the next release to be fixed. We will keep this thread updated. Thanks!

**Comment by radditude** [comment]
Thanks for reporting this! I was initially puzzled by this report - when building release binaries, we pass in the version as a linker flag, so it shouldn't matter how the version is defined in `version.go`. And indeed, the official binaries available on [releases.hashicorp.com](https://releases.hashicorp.com/terraform/1.4.4) report the correct version.

Unfortunately, the binary produced by `brew install terraform` is not the same as the one available through the releases site. HashiCorp maintains our own Homebrew formula at `hashicorp/tap/terraform`, which downloads binaries from the releases site, but the main `terraform` namespace is occupied by a [community-maintained formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/terraform.rb) which downloads the source code from Github and performs its own Go build process that doesn't match ours.

In short, we won't be able to resolve the problem with `brew install terraform` until the next release, but `brew install hashicorp/tap/terraform` will resolve the problem right away (and is the preferred way to install Terraform via Homebrew)

**Comment by wyardley** [comment]
Is the problem with the Ubuntu package related, and should I open a separate issue for it?

**Comment by radditude** [comment]
The Ubuntu/apt issue seems to be separate, unfortunately! If you wouldn't mind opening a separate issue, that would be very helpful for making sure it gets tracked and resolved properly.

**Comment by wyardley** [comment]
Thanks. I figured it was probably related to the build process changes, but not necessarily the same cause. I've opened #32964 for that

**Comment by benlangfeld** [comment]
I think this issue justifies cutting a 1.4.5 release just to fix it, since it is a significant inconvenience for those who pin the TF version for cross-environment consistency.

See also https://github.com/Homebrew/homebrew-core/pull/127125

**Comment by crw** [comment]
Just as an FYI and workaround, `brew install hashicorp/tap/terraform` should be working.

There exists two Terraform Homebrew formulae: the HashiCorp official `hashicorp/tap/terraform`, which downloads binaries from the releases site, and a community-maintained formula at `terraform`, which downloads the source code from GitHub and builds from source. 

We are migrating to a new build system, and the officially built binaries had the correct version passed in as linker args when they were built, but the unofficial tap falls back on the version defined in the code. Due to an issue with the new release process, that version is incorrect. In the future, this issue should be corrected so future versions of the `terraform` formula should build correctly.

**Comment by y0rune** [comment]
I have the same problem when I tried to install it via `go install`
```
yorune@PC /tmp $ GO111MODULE=on go install github.com/hashicorp/terraform@latest
yorune@PC /tmp $ terraform --version
Terraform v1.4.3-dev
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v4.48.0

Your version of Terraform is out of date! The latest version
is 1.4.4. You can update by downloading from https://www.terraform.io/downloads.html
```

**Comment by crw** [comment]
@y0rune Yes, this is expected and is the same problem that hit the unofficial brew package.

**Comment by brsolomon-deloitte** [comment]
One motivation for a 1.4.5 patch sooner rather than later is that this breaks https://github.com/hashicorp/tfc-getting-started/blob/main/scripts/setup.sh:

```shell
$ ./scripts/setup.sh
...

Press any key to continue (ctrl-c to quit):

Creating an organization and workspace...
An error occurred: Terraform version "1.4.3-dev" does not resolve
to a version of Terraform present in Terraform Cloud,
or this organization is not permitted to use it.
```
